.github/workflows/CI.yml

```diff
@@ -0,0 +1,65 @@
+name: full_tests
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
+  schedule:
+    # nightly tests, 2 am
+    - cron: "0 2 * * *"
+
```

Comment:
Worth adding a concurrency section? (i.e. something like this: https://github.com/MDAnalysis/mdanalysis/blob/5bf6359a821c085a1039ca414ce51ca2820722fb/.github/workflows/gh-ci.yaml#L12-L17)

Reply:
Yes that is perfect and I will start adding that to other projects, thanks!